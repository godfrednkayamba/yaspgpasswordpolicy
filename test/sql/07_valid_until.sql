LOAD 'yaspgpp';
--
--reset all settings
--
SET yaspgpp.username_min_length TO DEFAULT;
SET yaspgpp.username_min_special TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_digit TO DEFAULT;
SET yaspgpp.username_contain_password TO DEFAULT;
SET yaspgpp.username_ignore_case TO DEFAULT;
SET yaspgpp.username_contain TO DEFAULT;
SET yaspgpp.username_not_contain TO DEFAULT;
SET yaspgpp.username_min_repeat TO DEFAULT;
SET yaspgpp.password_min_length TO DEFAULT;
SET yaspgpp.password_min_special TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_digit TO DEFAULT;
SET yaspgpp.password_contain_username TO DEFAULT;
SET yaspgpp.password_ignore_case TO DEFAULT;
SET yaspgpp.password_contain TO DEFAULT;
SET yaspgpp.password_not_contain TO DEFAULT;
SET yaspgpp.password_min_repeat TO DEFAULT;
SET yaspgpp.password_reuse_history = 0;
SET yaspgpp.password_reuse_interval = 0;

-- VALID UNTIL clause checks
SET yaspgpp.password_valid_until TO 4;
SET yaspgpp.password_valid_max TO 0;
-- fail, the VALID UNTIL clause must be present
CREATE USER aaa PASSWORD 'DummY';
-- Success, the VALID UNTIL clause is present and respect the delay
CREATE USER aaa PASSWORD 'DummY' VALID UNTIL '2050-01-01 00:00:00';
-- fail, the VALID UNTIL clause does not respect the delay
ALTER USER aaa PASSWORD 'DummY2' VALID UNTIL '2022-01-01 00:00:00';
SET yaspgpp.password_valid_max TO 180;
-- fail, the VALID UNTIL clause can not exceed a maximum of 180 days
ALTER USER aaa PASSWORD 'DummY2' VALID UNTIL '2050-01-01 00:00:00';
-- Clear the user
DROP USER aaa;
-- fail, the VALID UNTIL clause can not exceed a maximum of 180 days
CREATE USER aaa PASSWORD 'DummY2' VALID UNTIL '2050-01-01 00:00:00';
SET yaspgpp.password_valid_until to 60;
SET yaspgpp.password_reuse_interval to 15;
SET yaspgpp.password_reuse_history to 4;
CREATE role yaspgpp_test with login password 'password'; 
-- History must be empty
SELECT count(*), '0' AS "expected" FROM pg_password_history ;
