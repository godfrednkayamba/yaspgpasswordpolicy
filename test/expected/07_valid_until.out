LOAD 'yaspgpasswordpolicy';
--
--reset all settings
--
SET yaspgpasswordpolicy.username_min_length TO DEFAULT;
SET yaspgpasswordpolicy.username_min_special TO DEFAULT;
SET yaspgpasswordpolicy.username_min_upper TO DEFAULT;
SET yaspgpasswordpolicy.username_min_upper TO DEFAULT;
SET yaspgpasswordpolicy.username_min_digit TO DEFAULT;
SET yaspgpasswordpolicy.username_contain_password TO DEFAULT;
SET yaspgpasswordpolicy.username_ignore_case TO DEFAULT;
SET yaspgpasswordpolicy.username_contain TO DEFAULT;
SET yaspgpasswordpolicy.username_not_contain TO DEFAULT;
SET yaspgpasswordpolicy.username_min_repeat TO DEFAULT;
SET yaspgpasswordpolicy.password_min_length TO DEFAULT;
SET yaspgpasswordpolicy.password_min_special TO DEFAULT;
SET yaspgpasswordpolicy.password_min_upper TO DEFAULT;
SET yaspgpasswordpolicy.password_min_upper TO DEFAULT;
SET yaspgpasswordpolicy.password_min_digit TO DEFAULT;
SET yaspgpasswordpolicy.password_contain_username TO DEFAULT;
SET yaspgpasswordpolicy.password_ignore_case TO DEFAULT;
SET yaspgpasswordpolicy.password_contain TO DEFAULT;
SET yaspgpasswordpolicy.password_not_contain TO DEFAULT;
SET yaspgpasswordpolicy.password_min_repeat TO DEFAULT;
SET yaspgpasswordpolicy.password_reuse_history = 0;
SET yaspgpasswordpolicy.password_reuse_interval = 0;
-- VALID UNTIL clause checks
SET yaspgpasswordpolicy.password_valid_until TO 4;
SET yaspgpasswordpolicy.password_valid_max TO 0;
-- fail, the VALID UNTIL clause must be present
CREATE USER aaa PASSWORD 'DummY';
ERROR:  require a VALID UNTIL option
-- Success, the VALID UNTIL clause is present and respect the delay
CREATE USER aaa PASSWORD 'DummY' VALID UNTIL '2050-01-01 00:00:00';
-- fail, the VALID UNTIL clause does not respect the delay
ALTER USER aaa PASSWORD 'DummY2' VALID UNTIL '2022-01-01 00:00:00';
ERROR:  the VALID UNTIL option must have a date older than 4 days
SET yaspgpasswordpolicy.password_valid_max TO 180;
-- fail, the VALID UNTIL clause can not exceed a maximum of 180 days
ALTER USER aaa PASSWORD 'DummY2' VALID UNTIL '2050-01-01 00:00:00';
ERROR:  the VALID UNTIL option must NOT have a date beyond 180 days
-- Clear the user
DROP USER aaa;
-- fail, the VALID UNTIL clause can not exceed a maximum of 180 days
CREATE USER aaa PASSWORD 'DummY2' VALID UNTIL '2050-01-01 00:00:00';
ERROR:  require a VALID UNTIL option with a date beyond 180 days
SET yaspgpasswordpolicy.password_valid_until to 60;
SET yaspgpasswordpolicy.password_reuse_interval to 15;
SET yaspgpasswordpolicy.password_reuse_history to 4;
CREATE role yaspgpasswordpolicy_test with login password 'password'; 
ERROR:  require a VALID UNTIL option
-- History must be empty
SELECT count(*), '0' AS "expected" FROM pg_password_history ;
 count | expected 
-------+----------
     0 | 0
(1 row)

